lifting-state-up/passing-down-event-handlers.md

```diff
@@ -0,0 +1,250 @@
+
+## Create a Function to Update `studentData` in `App`
+
+We can create as many functions and helper functions inside our React components as we want!
+
+In `App`, let's create a function named `updateStudentData`. It's responsible for taking in the updated data for one student, and updating `studentData` on state.
+
+Since this function updates _one_ student, it needs to receive the updated student data. This function should accept a parameter, `updatedStudentData`.
+
+This function should go inside the `App` component, after `studentData` is defined, and before the `return` statement.
+
+<!-- prettier-ignore-start -->
+```js
+  const updateStudentData = updatedStudent => {
+    const students = studentData.map(student => {
+      if (student.id === updatedStudent.id) {
+        return updatedStudent;
+      } else {
+        return student;
+      }
+    });
+
+    setStudentData(students);
+  }
+```
+<!-- prettier-ignore-end -->
+
+| <div style="min-width:300px;"> Piece of Code </div> | Notes                                                                                                                                                                |
+| --------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `const updateStudentData = `                        | We're creating a function named `updateStudentData`.                                                                                                                 |
+| `updatedStudent`                                    | This function accepts one argument: an object that holds updated student data. We expect this object to have the key `id`, at the very least.                        |
+| `const students = ;`                                | For this function, we'll create a helper array, `students` to contain the updated student data.                                                                      |
+| `studentData.map(...)`                              | We'll map over `studentData`...                                                                                                                                      |
+| `student => {...}`                                 | Inside `map`, we'll reference each item with `student`...                                                                                                            |
+| `if (student.id === updatedStudent.id) {`           | If we found the student that our `updatedStudent` is updating...                                                                                                     |
+| `return updatedStudent`                             | The item we should map to our `students` array is the updated student data!                                                                                          |
```

Comment:
"map to" is fine from a mathematical perspective, but just from how map works as a function, and the idea that we're building values to go into the new array, "into" sounds better there
```suggestion
| `return updatedStudent`                             | The item we should map into our `students` array is the updated student data!                                                                                          |
```